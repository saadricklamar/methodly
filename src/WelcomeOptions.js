import React, { Component } from 'react';
import './WelcomeOptions.scss';


class WelcomeOptions extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }


  }


  render() {
    return(
      <div className='welcome'>
        <header className='welcome-header'>
          <h1>Methodly: Where Socrates Lives Or Dies According To How Much You Know</h1>
        </header>
        <h3>Please Select A Path</h3>
        <section class='choices'>
          <input type='submit' value='String Methods' onClick={this.props.chooseGamePath}/>
          <input type='submit' value='Math Methods' onClick={this.props.chooseGamePath}/>
        </section>
      </div>
      )

  }




}


 export default WelcomeOptions;