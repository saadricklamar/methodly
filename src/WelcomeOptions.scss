@import './variables.scss';

.welcome {
  background: linear-gradient( rgba(0, 0, 0, 0.2), rgba(0, 0, 0, 0.2) ), url(./images/Ancient.jpg) no-repeat center center fixed;
 -webkit-background-size: cover;
 -moz-background-size: cover;
 -o-background-size: cover;
 background-size: cover;
 display: flex;
 flex-direction: column;
 justify-content: center;
 align-items: center;
 height: 100vh;
 font-weight: 400;
 color: $primary-color;
}

.welcome-header {
  position: relative;
  bottom: 300px;
  right: 200px;
  font-family: $primary-font;
}

h3 {
  font-family: $primary-font;
  font-size: 50px;
  margin-top: 0;
}

.choices {
  display: flex;
  justify-content: space-between;
  input {
    width: 200px;
    height: 100px;
    background-color: $primary-background-color;
    border-radius: 5px;
    margin: 20px;
    cursor: $primary-pointer;
    font-family: $primary-font;
    font-size: 30px;
    color: $primary-color;
    &:hover {
      background-color: $primary-hover-color;
    }
  }
}



