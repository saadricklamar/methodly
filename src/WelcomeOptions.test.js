import React from 'react';
import WelcomeOptions from './WelcomeOptions.js';
import { shallow } from 'enzyme';

const mockFunc = jest.fn();

describe('WelcomeOptions', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<WelcomeOptions 
                      chooseGamePath={mockFunc}
                      chooseGamePath={mockFunc}
                      />)
});
it('Should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
});
