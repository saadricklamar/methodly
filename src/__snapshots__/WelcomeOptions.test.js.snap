// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeOptions Should match the snapshot 1`] = `
<div
  className="welcome"
>
  <header
    className="welcome-header"
  >
    <h1>
      Methodly: Where Socrates Lives Or Dies According To How Much You Know
    </h1>
  </header>
  <h3>
    Please Select A Path
  </h3>
  <section
    class="choices"
  >
    <input
      onClick={[MockFunction]}
      type="submit"
      value="String Methods"
    />
    <input
      onClick={[MockFunction]}
      type="submit"
      value="Math Methods"
    />
  </section>
</div>
`;
